### PR TITLE
[Paddle]Add chapter_deep-learning-computation-update

### PR DESCRIPTION
**Pull request by w5688414**

### CODE REVIEW

**Comment by w5688414** [comment]
@AnirudhDagar @cheungdaven This is the deep-learning-computation chapter, can you review thie chapter? Thanks

File: chapter_deep-learning-computation/custom-layer.md [hunk 1]
```diff
@@ -167,6 +193,19 @@ class MyDense(tf.keras.Model):
...
         return tf.nn.relu(linear)
 ```
+
+```{.python .input}
+#@tab paddle
+class MyLinear(nn.Layer):
+    def __init__(self, in_units, units):
+        super().__init__()
+        self.weight = paddle.create_parameter(shape=(in_units, units), dtype='float32')
+        self.bias = paddle.create_parameter(shape=(units,), dtype='float32')
```
Comment (cheungdaven) [inline chapter_deep-learning-computation/custom-layer.md:203]: a blank line below

File: chapter_deep-learning-computation/model-construction.md [hunk 2]
```diff
@@ -321,6 +372,25 @@ class MySequential(tf.keras.Model):
         return X
 ```
 
+```{.python .input  n=3}
+#@tab paddle
+class MySequential(nn.Layer):
+    def __init__(self, *layers):
+        super(MySequential, self).__init__()
+        if len(layers) > 0 and isinstance(layers[0], tuple): # 如果传入的是一个tuple
```
Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:380]: use a single line for this comment

Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:380]: use a single line for this comment

File: chapter_deep-learning-computation/model-construction.md [hunk 3]
```diff
@@ -321,6 +372,25 @@ class MySequential(tf.keras.Model):
...
+            for name, layer in layers:
+                self.add_sublayer(name, layer)  #add_sublayer方法会将layer添加到self._sub_layers(一个tuple)
```
Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:382]: ditto

Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:382]: ditto

File: chapter_deep-learning-computation/model-construction.md [hunk 4]
```diff
@@ -471,8 +571,8 @@ net.initialize()
 net(X)
 ```
 
-```{.python .input}
-#@tab pytorch, tensorflow
+```{.python .input  n=7}
```
Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:574]: remove n=7

Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:574]: remove n=7

File: chapter_deep-learning-computation/model-construction.md [hunk 5]
```diff
@@ -534,6 +634,22 @@ chimera.add(FixedHiddenMLP())
...
 ```
 
+```{.python .input  n=8}
+#@tab paddle
+class NestMLP(nn.Layer):
+    def __init__(self):
+        super().__init__()
+        self.net = nn.Sequential(nn.Linear(20, 64), nn.ReLU(),
+                                 nn.Linear(64, 32), nn.ReLU())
+        self.linear = nn.Linear(32, 16)
```
Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:644]: I am wondering if paddle provides interface like lazylinear in pytorch.

Comment (cheungdaven) [inline chapter_deep-learning-computation/model-construction.md:644]: I am wondering if paddle provides interface like lazylinear in pytorch.

File: chapter_deep-learning-computation/use-gpu.md [hunk 6]
```diff
@@ -52,6 +52,23 @@
 我们通常希望模型的参数在GPU上。
 :end_tab:
 
+:begin_tab:`paddle`
+在PaddlePaddle中，每个数组都有一个设备（device），
```
Comment (cheungdaven) [inline chapter_deep-learning-computation/use-gpu.md:56]: 数组？ do you mean tensor? if so, i do not think this is a good translation.

Comment (cheungdaven) [inline chapter_deep-learning-computation/use-gpu.md:56]: 数组？ do you mean tensor? if so, i do not think this is a good translation.

**Review by cheungdaven — CHANGES_REQUESTED** [review]

**Comment by cheungdaven** [comment]
please also fix the conflicts, thanks!

File: chapter_deep-learning-computation/model-construction.md (hunk 5 above)
Comment (w5688414) [reply chapter_deep-learning-computation/model-construction.md:644]: PaddlePaddle don't provide lazy API like pytorch, I have submitted a issue for this API, https://github.com/PaddlePaddle/Paddle/issues/42703

Comment (w5688414) [reply chapter_deep-learning-computation/model-construction.md:644]: PaddlePaddle don't provide lazy API like pytorch, I have submitted a issue for this API, https://github.com/PaddlePaddle/Paddle/issues/42703

File: chapter_deep-learning-computation/use-gpu.md (hunk 6 above)
Comment (w5688414) [reply chapter_deep-learning-computation/use-gpu.md:56]: this is the pytorch version:
```
:begin_tab:`pytorch`
在PyTorch中，每个数组都有一个设备（device），
我们通常将其称为上下文（context）。
```
our description follows the pytorch description, I change it to Tensor now:

```
在PaddlePaddle中，每个张量都有一个设备（device），
我们通常将其称为上下文（context）。
默认情况下，所有变量和相关的计算都分配给CPU。
有时上下文可能是GPU。
当我们跨多个服务器部署作业时，事情会变得更加棘手。
.....
```

**Comment by w5688414** [comment]
@AnirudhDagar @cheungdaven I replied the question and fix the styleing issue, can you review thie chapter? Thanks

**Pull request merged**